firmware/common2015/drivers/kicker-board/KickerBoard.cpp

```diff
@@ -89,4 +90,69 @@ bool KickerBoard::flash(bool onlyIfDifferent, bool verbose) {
     return true;
 }
 
+/* this function enforces the design choice that each cmd must have an arg
+ * and return a value */
+uint8_t KickerBoard::send_to_kicker(const uint8_t cmd, const uint8_t arg,
+                                    bool verbose) {
+    chipSelect();
+    // Returns state (charging, not charging), but we don't care about that
+    uint8_t charge_resp = _spi->write(cmd);
+    // Should return the command we just sent
+    uint8_t command_resp = _spi->write(arg);
+    // Should return final response to full cmd, arg pair
+    uint8_t return_resp = _spi->write(BLANK);
+    chipDeselect();
+
+    if (verbose) {
+        printf("Kicker: CHG:%02X, CMD:%02X, RET:%02X\r\n", charge_resp,
+               command_resp, return_resp);
+        fflush(stdout);
+    }
+
+    return return_resp;
+}
+
+uint8_t KickerBoard::kick(uint8_t time) {
+    return send_to_kicker(KICK_CMD, time);
+}
+
+uint8_t KickerBoard::chip(uint8_t time) {
+    return send_to_kicker(CHIP_CMD, time);
+}
+
+uint8_t KickerBoard::read_voltage() {
+    return send_to_kicker(GET_VOLTAGE_CMD, BLANK);
+}
+
+uint8_t KickerBoard::charge() { return send_to_kicker(SET_CHARGE_CMD, ON_ARG); }
```

Comment:
I would split this into 3 lines instead of 1 so it matches the rest of the kicker board commands